build/tasks.ts

```diff
@@ -129,97 +130,43 @@ const buildWeb = register("build:web", async (runner) => {
-		await copyForDefaultExtensions();
-		runner.cwd = extDirPath;
-		const resp = await runner.execute(isWin ? "npx.cmd" : "npx", [isWin ? "gulp.cmd" : "gulp", "vscode-linux-x64", "--max-old-space-size=32384"]);
-		if (resp.exitCode !== 0) {
-			throw new Error(`Failed to build default extensions: ${resp.stderr}`);
-		}
-	}
-});
-
 const ensureInstalled = register("vscode:install", async (runner) => {
-	await ensureCloned();
+	runner.cwd = libPath;
 
-	runner.cwd = vscodePath;
-	const install = await runner.execute(isWin ? "yarn.cmd" : "yarn", []);
-	if (install.exitCode !== 0) {
-		throw new Error(`Failed to install vscode dependencies: ${install.stderr}`);
-	}
-});
-
-const ensureCloned = register("vscode:clone", async (runner) => {
-	if (fs.existsSync(vscodePath)) {
-		await ensureClean();
-	} else {
-		fse.mkdirpSync(libPath);
-		runner.cwd = libPath;
-		const clone = await runner.execute("git", ["clone", "https://github.com/microsoft/vscode", "--branch", vscodeVersion, "--single-branch", "--depth=1"]);
-		if (clone.exitCode !== 0) {
-			throw new Error(`Failed to clone: ${clone.exitCode}`);
+	if (fs.existsSync(vscodePath) && fs.existsSync(defaultExtensionsPath)) {
+		const pkgVersion = JSON.parse(fs.readFileSync(path.join(vscodePath, "package.json")).toString("utf8")).version;
+		if (pkgVersion === vscodeVersion) {
+			return;
 		}
 	}
 
-	runner.cwd = vscodePath;
-	const checkout = await runner.execute("git", ["checkout", vscodeVersion]);
-	if (checkout.exitCode !== 0) {
-		throw new Error(`Failed to checkout: ${checkout.stderr}`);
-	}
-});
+	fse.removeSync(libPath);
+	fse.mkdirpSync(libPath);
 
-const ensureClean = register("vscode:clean", async (runner) => {
-	runner.cwd = vscodePath;
+	await new Promise<void>((resolve, reject): void => {
+		https.get(vsSourceUrl, (res) => {
+			if (res.statusCode !== 200) {
+				return reject(res.statusMessage);
+			}
 
-	const status = await runner.execute("git", ["status", "--porcelain"]);
-	if (status.stdout.trim() !== "") {
-		const clean = await runner.execute("git", ["clean", "-f", "-d", "-X"]);
-		if (clean.exitCode !== 0) {
-			throw new Error(`Failed to clean git repository: ${clean.stderr}`);
-		}
-		const removeUnstaged = await runner.execute("git", ["checkout", "--", "."]);
-		if (removeUnstaged.exitCode !== 0) {
-			throw new Error(`Failed to remove unstaged files: ${removeUnstaged.stderr}`);
-		}
-	}
-	const fetch = await runner.execute("git", ["fetch", "--prune"]);
-	if (fetch.exitCode !== 0) {
-		throw new Error(`Failed to fetch latest changes: ${fetch.stderr}`);
-	}
+			res.pipe(tar.x({
+				C: libPath,
+			}).on("finish", () => {
+				resolve();
+			}).on("error", (err: Error) => {
+				reject(err);
+			}));
+		}).on("error", (err) => {
+			reject(err);
+		});
+	});
 });
 
 const ensurePatched = register("vscode:patch", async (runner) => {
 	if (!fs.existsSync(vscodePath)) {
 		throw new Error("vscode must be cloned to patch");
 	}
-	await ensureClean();
+	await ensureInstalled();
```

Comment:
Should we still keep the `git reset` that was inside `ensureClean`? Otherwise trying to build the binary again fails because the patch won't apply. Or maybe if the patch fails we log a notice and keep going anyway. That might be better so we can build the binary locally for testing with custom changes that we haven't put into the patch yet.

Reply:
Ah I agree. Will add.